p2p/switch.go

```diff
@@ -148,6 +149,7 @@ func (sw *Switch) IsListening() bool {
 // NOTE: Not goroutine safe.
 func (sw *Switch) SetNodeInfo(nodeInfo NodeInfo) {
 	sw.nodeInfo = nodeInfo
+	sw.nodePublicAddr = nodeInfo.ListenAddr
```

Comment:
should this go in `nodeInfo`?

Reply:
We keep track of our addresses in the AddrBook. See `OurAddress` methods